src/NumericalAlgorithms/LinearSolver/Convergence.hpp

```diff
@@ -0,0 +1,188 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/none.hpp>
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <iosfwd>
+
+#include "NumericalAlgorithms/LinearSolver/IterationId.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace LinearSolver {
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief Criteria that determine the linear solve has converged
+ *
+ * \details Most criteria are based on the residual magnitude
+ * \f$|r_k|=|b-Ax_k|\f$, where \f$x_k\f$ denotes the numerical solution after
+ * completion of iteration \f$k\f$ (see the \ref LinearSolverGroup
+ * documentation, `LinearSolver::Tags::Residual` and
+ * `LinearSolver::Tags::Magnitude`).
+ *
+ * The following criteria are implemented, ordered from highest to lowest
+ * priority:
+ *
+ * - AbsoluteResidual: Matches if the residual has reached this magnitude.
+ * - RelativeResidual: Matches if the residual has decreased by this factor,
+ * relative to the start of the first linear solver iteration.
+ * - MaxIterations: Matches if the number of iterations exceeds this limit.
+ *
+ * \note that the smallest possible residual magnitude the linear solver can
```

Comment:
Start a new sentence after each `\note`.  The rendered documentation does not format in a way that "Note" can be read as part of the sentence.  (It is completely omitted in some cases, even.)